### ISSUE TITLE
Options for doing GETs after SETs

### ISSUE THREAD

**Issue by GoogleCodeExporter** [OpenZWave/open-zwave#154]
```
Many people have requested more flexibility in how the library performs the 
Gets after Sets in its value classes. Currently the code will always request an 
update as this is used to do notification to the user programs as well as keep 
the library in sync with the device. It is necessary to do this as sometimes 
the device doesn't process or accept the Set so this is the only way to track 
the real value.

Now someone wants to not bother with the Gets at all in an effort to perform 
high speed Sets. More like an unreliable Set.

There is also a need to address different dimmer devices that take differing 
amounts of time to update the device. The initial Get will almost always report 
the incorrect value. This is becoming a FAQ on the mailing list and something 
the library can handle better.

This mailing list discussion thread has lots of detail.

https://groups.google.com/forum/#!topic/openzwave/JX0O1BnBy2g

Part of the feature addition will require a Manager object interface so each 
value can determine how the set should be performed. It might also be useful to 
augment the set interface to be able to do it per call.
```

Original issue reported on code.google.com by `glsatz` on 22 Jan 2013 at 6:50

**Comment by stromnet** [comment]
Hi,
would like to wake this question up again. I've got a bunch of Z-wave switch & dimmers in the same & adjecent rooms, when sending "OFF" or "ON" to all of these at the same time I get the irritating "wave" of changes. Sometimes everything changes state within a second, but sometimes one or two nodes may get stuck for a second or so before it reacts. 
To minimize this time I'd want to down-prioritize, or disable, the GET-after-SET request. 

In my case the slaves (Tellstick TZWP-100) always send a SwitchBinary report at or right after the SET/GET is issued, or for dimmers (Fibaro FGD-211) one or several SwitchMultiLevel report, so the GET is not really required. But from the mailinglist I understand that this is not always the case? Even though it seems active polling is required anyway in such cases, since dimmers may take longer than the initial GET..


A solution which would be nice from at least my point of view, partially based on what I've read in mail list, and partially based on my quite limited OpenZWave internals knowledge (thus, it might be hard to do):

* Move the automated GET to separate queue, lower prio than Send but higher than Poll
* If a new automated GET is about to be enqueued, check if the device already has one pending on this new queue. If so, discard the new one.
* When a MultiValue/SwitchReport is received after the SET has been executed, clear any pending low-prio GET (we just got a fresh value, no need to GET-after-SET it)


Any updated ideas on this subject now 3 years after initial ticket & mail discussion?
